acra-censor/handlers/blacklist_handler.go

```diff
@@ -273,101 +267,78 @@ func checkSinglePatternMatch(queryNodes []sqlparser.SQLNode, patternNodes []sqlp
+	matchDetected := false
+	if len(patternNodes) != len(queryNodes) {
+		return false
+	}
+	for index, patternNode := range patternNodes {
+		if index == 0 || reflect.DeepEqual(patternNode, queryNodes[index]) {
+			continue
+		}
+		if patternNodeColName, ok := patternNode.(*sqlparser.ColName); ok && patternNodeColName != nil {
+			if strings.EqualFold(patternNodeColName.Name.String(), ColumnConfigPlaceholderReplacer) {
+				matchDetected = true
 			} else {
-				return true, false
+				return false
 			}
 		}
 	}
-	return false, false
+	return matchDetected
 }
 
 //handle SELECT a, b from t %%WHERE%% pattern
-func handleSelectWherePattern(queryNodes, patternNodes []sqlparser.SQLNode) (patternDetected bool, matchOccurred bool) {
-	var emptySelect *sqlparser.Select
-	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(emptySelect) && reflect.TypeOf(patternNodes[0]) == reflect.TypeOf(emptySelect) {
-		patternDetected := false
-		patternHasWhereNode := false
-		queryHasWhereNode := false
-
-		greaterLength := len(queryNodes)
-		if greaterLength < len(patternNodes) {
-			greaterLength = len(patternNodes)
+func handleSelectWherePattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
+	patternWhereDetected := false
+	queryWhereDetected := false
+	for index, patternNode := range patternNodes {
+		if index == 0 || reflect.DeepEqual(queryNodes[index], patternNode) {
+			continue
 		}
-		//start from first node because zero node represents all query
-		index := 1
-		for index < greaterLength {
-			if index < len(queryNodes) && index < len(patternNodes) {
-				//if nodes are not equal, the only case when query matches pattern is if non-equal nodes are both 'where' nodes and pattern's where is replacer of placeholder (WhereConfigPlaceholderReplacerPart2)
-				if !reflect.DeepEqual(queryNodes[index], patternNodes[index]) {
-					if wherePatternNode, ok := patternNodes[index].(*sqlparser.Where); ok && wherePatternNode != nil {
-						if strings.EqualFold(sqlparser.String(wherePatternNode.Expr), WhereConfigPlaceholderReplacerPart2) {
-							patternDetected = true
-						}
-						patternHasWhereNode = true
-					}
-					if whereQueryNode, ok := queryNodes[index].(*sqlparser.Where); ok && whereQueryNode != nil {
-						queryHasWhereNode = true
-					}
-					//both pattern and query has non-nil 'where' node
-					if patternHasWhereNode == queryHasWhereNode && patternHasWhereNode == true {
-						return patternDetected, true
-					}
-
-					return patternDetected, false
-				}
-				index++
-			}
+		if patternWhereNode, ok := patternNode.(*sqlparser.Where); ok && strings.EqualFold(sqlparser.String(patternWhereNode.Expr), WhereConfigPlaceholderReplacerPart2) {
+			patternWhereDetected = true
+		}
+		if queryWhereNode, ok := queryNodes[index].(*sqlparser.Where); ok && queryWhereNode != nil {
+			queryWhereDetected = true
+		}
+		if patternWhereDetected == queryWhereDetected && patternWhereDetected {
```

Comment:
true == true && true ?)
better to leave only `patternWhereDetected && queryWhereDetected`, because it's true only when both are true